### ISSUE TITLE
pydantic model validator triggers N805: First argument of a method should be named `self`

### ISSUE THREAD

**Issue by hofrob** [charliermarsh/ruff#2151]
Sorry to bother you guys with this again (#769), but there may be a regression here (or I'm too stupid  to find out what I'm doing wrong).

Reproduce:

* create a new project with the current version of ruff (0.0.231 at the time of writing)
* create a python file with a pydantic model with a validator
* run `ruff -n --select N805 n805.py`

Contents of `n805.py` (which looks exactly like the test case):

```python
import pydantic


class TestN805(pydantic.BaseModel):
    first: str
    second: str

    @pydantic.validator("first")
    def min_length(cls, value: str) -> str:
        pass
```

Result: 

```bash
n805.py:9:20: N805 First argument of a method should be named `self`
```

**Comment by charliermarsh** [comment]
No prob! Can you try adding this to your `pyproject.toml`?

```toml
[tool.ruff.pep8-naming]
# Allow Pydantic's `@validator` decorator to trigger class method treatment.
classmethod-decorators = ["classmethod", "pydantic.validator"]
```

There's an explanation [here](https://github.com/charliermarsh/ruff#pep8-naming) in the README.

**Comment by charliermarsh** [comment]
(This tells Ruff that methods annotated with `@pydantic.validator` (or `from pydantic import validator; @validator`) should be considered class methods.)

**Comment by hofrob** [comment]
Oh, so I have to tell ruff which methods are considered as class methods!

Thanks a lot for the quick reply! And for this insanely useful tool!

**Comment by charliermarsh** [comment]
Awesome, glad it's working for you :)

**Comment by Bobronium** [comment]
Comming here from https://github.com/PyCQA/pep8-naming/issues/169

Couldn't get it to work with `"validator"` added to config, but `"pydantic.validator"`, as suggested here, worked.

**Comment by luminoso** [comment]
The updated syntax for ruff and pydantic v2 is:

```toml
[tool.ruff.lint.pep8-naming]
# Allow Pydantic's `@validator` decorator to trigger class method treatment.
classmethod-decorators = ["classmethod", "pydantic.field_validator"]
```

**Comment by FBosler** [comment]
What about model_validator? it behaves differently for `mode="before"` vs `mode="after"` one should be a class method, one an instance method. I additionally decorated the ones that are classmethods with the classmethod decorator ... but don't love it
